service/controller/handler/values/handler.go

```diff
@@ -156,19 +161,51 @@ func (h *Handler) generateConfig(ctx context.Context, installation, namespace, a
 		}
 	}
 
-	gen, err := generator.New(&generator.Config{
+	gen, err := generator.New(generator.Config{
 		Fs:               store,
 		DecryptTraverser: h.decryptTraverser,
-		ProjectVersion:   h.projectVersion,
+		Installation:     h.installation,
 	})
 	if err != nil {
 		return nil, nil, microerror.Mask(err)
 	}
 
-	configmap, secret, err = gen.GenerateConfig(ctx, installation, app, namespace, ref)
+	meta := metav1.ObjectMeta{
+		Name:      generateResourceName(app, ref),
+		Namespace: namespace,
+
+		Annotations: map[string]string{
+			key.ReleaseNameAnnotation:      generateResourceName(app, ref),
+			key.ReleaseNamespaceAnnotation: namespace,
+		},
+		Labels: map[string]string{
+			key.KubernetesManagedByLabel:  "Helm",
+			label.AppKubernetesName:       app,
+			label.ConfigControllerVersion: h.projectVersion,
+			label.ManagedBy:               project.Name(),
+		},
+	}
+
+	configmap, secret, err = gen.GenerateConfig(ctx, app, meta)
 	if err != nil {
 		return nil, nil, microerror.Mask(err)
 	}
 
 	return configmap, secret, nil
 }
+
+var (
+	multipleDashPattern     = regexp.MustCompile("-{2,}")
+	invalidCharacterPattern = regexp.MustCompile("[^a-z0-9]+")
+)
+
+func generateResourceName(elements ...string) string {
```

Comment:
I didn't bother to re-add tests because this whole handler is supposed to go away.